aiida/orm/nodes/data/array/cube.py

```diff
@@ -0,0 +1,246 @@
+        voxel = np.empty((3, 3))
+        for i in range(3):
+            nmb, x, y, z = [float(s) for s in next(fiter).split()]
+            shape[i] = int(nmb)
+            voxel[i] = np.array([x, y, z])
+
+        self.voxel = voxel
+
+        # Atomic numbers, charges and positions.
+        numbers = np.empty(natoms, int)
+        charges = np.empty(natoms, int)
+        positions = np.empty((natoms, 3))
+        for i in range(natoms):
+            line = next(fiter).split()
+            numbers[i] = int(line[0])
+            charges[i] = float(line[1])
+            positions[i] = [float(s) for s in line[2:]]
+        self.atomic_numbers = numbers
+        self.atomic_charges = charges
+        self.atomic_coordinates = positions
+
+        # Data mesh.
+        data = np.empty(shape[0] * shape[1] * shape[2], dtype=float)
+        cursor = 0
+        for i, line in enumerate(fiter):
+            l_split = line.split()
+            data[cursor:cursor + len(l_split)] = l_split
+            cursor += len(l_split)
+        self.data = data.reshape(shape)
+
+        if data_units:
+            self.data_units = data_units
+
+    def everything_ok(self):
+        """Check if the object contains all the data to construct a valid cube file."""
+        all_good = True
+        for attr in ['comment', 'voxel', 'origin', 'atomic_numbers', 'atomic_charges', 'atomic_coordinates', 'data']:
+            try:
+                getattr(self, attr)
+            except (AttributeError, KeyError):
+                all_good = False
+                print(f'The attribute `{attr}` is not set.')
+        return all_good
+
+    def export(self, path, fileformat='cube', overwrite=False, **kwargs):
+        """Export data to the selected file."""
+
+        if not self.everything_ok():
+            raise ValueError("Can't export the cube file.")
+
+        if fileformat == 'cube':
+            content = self._write_gaussian_cube()
+
+        if path:
+            if overwrite or not os.path.isfile(path):
+                with open(path, 'w') as fobj:
+                    fobj.write(content)
+            else:
+                raise FileExistsError(
+                    "Can't write to an existing file, please set `overwrite=True` to avoid this exception."
+                )
+        else:
+            sys.stdout.write(content)
+
+    def _write_gaussian_cube(self):
+        """Write gaussian cube data into a string."""
+
+        # Comment.
+        filecontent = '\n'.join(self.comment) + '\n'
+        natoms = self.atomic_numbers.shape[0]
+
+        # Number of atoms and origin.
+        filecontent += '{:5d} {:12.6f} {:12.6f} {:12.6f}\n'.format(
+            natoms, self.origin[0], self.origin[1], self.origin[2]
+        )
+
+        # Data shape and voxel.
+        for i in range(3):
+            filecontent += '{:5d} {:12.6f} {:12.6f} {:12.6f}\n'.format(
+                self.data.shape[i], self.voxel[i][0], self.voxel[i][1], self.voxel[i][2]
```

Comment:
This is not very efficient, you are calling `self.data` in each iteration (and also further down in the function) which causes a database hit every time. Best to get the data once and assign to local variable. Same for other data that are stored as attributes.